### PR TITLE
[SPARK-19349][DStreams]improve resource ready check to avoid multiple receivers to be scheduled on the same node.

### PR DESCRIPTION
**Pull request by uncleGen**
## What changes were proposed in this pull request?

remove related TODO

Currently, we can only ensure registered resource satisfy the "spark.scheduler.minRegisteredResourcesRatio". But if "spark.scheduler.minRegisteredResourcesRatio" is set too small, receivers may still be scheduled to few nodes. In fact, we can give once more chance to wait for sufficient resource to schedule receiver evenly.

## How was this patch tested?

update ut

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala [hunk 1]
```diff
@@ -422,16 +423,36 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
...
-   * "spark.scheduler.maxRegisteredResourcesWaitingTime" rather than running a dummy job.
+   * This only occurs when set too small "spark.scheduler.minRegisteredResourcesRatio".
    */
-  private def runDummySparkJob(): Unit = {
+  private def checkResourceReady(): Unit = {
+    val pollTime = 100
+    val checkingStarted = System.currentTimeMillis()
+    val onceMoreWaitingTimeMs =
+      ssc.sparkContext.conf.getTimeAsMs("spark.scheduler.maxRegisteredResourcesWaitingTime", "30s")
+
```
Comment (uncleGen) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala]: here use "spark.scheduler.maxRegisteredResourcesWaitingTime", need specific config？

**Comment by uncleGen** [comment]
retest this please.

**Comment by uncleGen** [comment]
Working on the test failure.

**Comment by uncleGen** [comment]
cc @zsxwing

**Comment by zsxwing** [comment]
I don't think all cases need to wait for all resources ready. That's why `spark.scheduler.minRegisteredResourcesRatio` is added to allow the user changing the behavior.

**Comment by uncleGen** [comment]
@zsxwing That is right, this pr is the improvement for the previous one. There is no much performance regression, i.e. only wait for "spark.scheduler.maxRegisteredResourcesWaitingTime" for resource ready (enough executors for receivers or satisfy the "minRegisteredResourcesRatio")

Any suggestion is appreciated.

**Comment by uncleGen** [comment]
cc @tdas also

**Comment by zsxwing** [comment]
I don't like copying codes from Spark core. Maybe let's still keep using a dummy Spark job.

**Pull request closed without merging**